### PR TITLE
Use dlib_vector_length implemented on pdlib 1.0.2 

### PR DESCRIPTION
**Pull request by matiasdelellis**
See https://github.com/matiasdelellis/pdlib/pull/2

In a small benchmark with 1963 descriptors:

OLD php: 0.012276937961578 Sec.
New native: 0.0022540950775146 Sec.

This is 80% faster than our PHP class.

### CODE REVIEW

File: lib/BackgroundJob/Tasks/CreateClustersTask.php [hunk 1]
```diff
@@ -232,19 +232,38 @@ private function getNewClusters(array $faces): array {
 		$sensitivity = $this->settingsService->getSensitivity();
 		$min_confidence = $this->settingsService->getMinimumConfidence();
 		$edges = array();
-		for ($i = 0, $face_count1 = count($faces); $i < $face_count1; $i++) {
-			$face1 = $faces[$i];
-			if ($face1->confidence < $min_confidence) {
-				$edges[] = array($i, $i); // fixme: Should we create an single group? o just ignore.
-				continue;
+
+		if (version_compare(phpversion('pdlib'), '1.0.2', '>=')) {
```
Comment (matiasdelellis) [inline lib/BackgroundJob/Tasks/CreateClustersTask.php:236]: The benchmark would be to add `$startTime = microtime (true);`

File: lib/BackgroundJob/Tasks/CreateClustersTask.php [hunk 2]
```diff
@@ -232,19 +232,38 @@ private function getNewClusters(array $faces): array {
...
+					continue;
+				}
+				for ($j = $i, $face_count2 = count($faces); $j < $face_count2; $j++) {
+					$face2 = $faces[$j];
+					$distance = dlib_vector_length($face1->descriptor, $face2->descriptor);
+
+					if ($distance < $sensitivity) {
+						$edges[] = array($i, $j);
+					}
+				}
```
Comment (matiasdelellis) [inline lib/BackgroundJob/Tasks/CreateClustersTask.php:250]: ..and:

```
$stopTime = microtime (true);
print "\n\nNative length:" . ($stopTime-$startTime)/1000 . " Sec.\n\n"; 
```

File: lib/BackgroundJob/Tasks/CreateClustersTask.php [hunk 3]
```diff
@@ -232,19 +232,38 @@ private function getNewClusters(array $faces): array {
...
 			}
-			for ($j = $i, $face_count2 = count($faces); $j < $face_count2; $j++) {
-				$face2 = $faces[$j];
-				// todo: can't this distance be a method in $face1->distance($face2)?
-				$distance = $euclidean->distance($face1->descriptor, $face2->descriptor);
+		} else {
```
Comment (matiasdelellis) [inline lib/BackgroundJob/Tasks/CreateClustersTask.php:252]: And the same here to compare..
I ran it many times, and it's stable, but  more importantly, the resulting custers are the same ..

**Comment by stalker314314** [comment]
Awesome! I was under impression that `dlib_chinese_whispers` was largest offender! This will speed up my calculation, it takes me 30 min and more to calculate clusters:o

**Comment by matiasdelellis** [comment]
Keep in mind that this improves the obtaining of the edges.. But the same chinese_whisper is still used. :wink: 

Well, when you test it, please share the results in your original report.. https://github.com/matiasdelellis/facerecognition/issues/105

**Pull request merged**